torch_xla/distributed/cluster.py

```diff
@@ -130,16 +132,59 @@ def _tpu_with_health(tpu_name):
 
   def wait_for_healthy_service(self):
 
-    def wait_for_healthy_worker(tpu_name):
+    def wait_for_healthy_service_worker(tpu_name):
       ctc = cloud_tpu_client.Client(tpu=tpu_name)
       ctc.wait_for_healthy()
 
     tpus = self.list_tpus_with_health('UNHEALTHY_MAINTENANCE')
+    if not tpus:
+      return
     with futures.ThreadPoolExecutor(max_workers=len(tpus)) as executor:
-      results = executor.map(wait_for_healthy_worker, tpus)
+      results = executor.map(wait_for_healthy_service_worker, tpus)
       for result in results:
         continue  # Only iterating to re-raise any worker errors.
 
+  def wait_for_healthy_client(
+      self, dist_executor, timeout_s=1200, interval_s=10):
```

Comment:
In Python times are usually implicitly in seconds (double floating point), so I think we can drop _s.